src/EditorFeatures/Core/Implementation/MissingImportTracking/MissingImportTrackingTaggerProvider.PasteCommandHandler.cs

```diff
@@ -0,0 +1,99 @@
+            public PasteCommandHandler(IThreadingContext threadingContext, ILightBulbBroker2 lightBulbBroker, ITextBufferAssociatedViewService textBufferAssociatedViewService, ISuggestedActionCategoryRegistryService suggestedActionCategoryRegistry)
+                : base(threadingContext)
+            {
+                _lightBulbBroker = lightBulbBroker;
+                _textBufferAssociatedViewService = textBufferAssociatedViewService;
+                _suggestedActionCategoryRegistry = suggestedActionCategoryRegistry;
+            }
+
+            public VSCommanding.CommandState GetCommandState(PasteCommandArgs args, Func<VSCommanding.CommandState> nextCommandHandler)
+            {
+                return nextCommandHandler();
+            }
+
+            public void ExecuteCommand(PasteCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+            {
+                var caretPosition = args.TextView.GetCaretPoint(args.SubjectBuffer);
+
+                nextCommandHandler();
+
+                if (!args.SubjectBuffer.CanApplyChangeDocumentToWorkspace())
+                {
+                    return;
+                }
+
+                var trackingSpan = caretPosition.Value.Snapshot.CreateTrackingSpan(caretPosition.Value.Position, 0, SpanTrackingMode.EdgeInclusive);
+
+                var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                if (document == null)
+                {
+                    return;
+                }
+
+                var addMissingImportsService = document.GetLanguageService<IAddMissingImportsFeatureService>();
+                var diagnostics = addMissingImportsService.GetDiagnosticsAsync(document, executionContext.OperationContext.UserCancellationToken).GetAwaiter().GetResult();
+
+                if (diagnostics.Length == 0)
+                {
+                    return;
+                }
+
+                RegisterSpanWithMissingImports(document, trackingSpan);
+
+                var associatedTextViews = _textBufferAssociatedViewService.GetAssociatedTextViews(trackingSpan.TextBuffer);
+                var textView = associatedTextViews.FirstOrDefault(); // TODO: Don't be so optimistic
+
+                if (textView == null)
+                {
+                    return;
+                }
+
+                var endPoint = trackingSpan.GetEndPoint(trackingSpan.TextBuffer.CurrentSnapshot);
+                var endTrackingSpan = trackingSpan.TextBuffer.CurrentSnapshot.CreateTrackingSpan(endPoint.Position, 0, SpanTrackingMode.EdgeInclusive);
+                var endTrackingPoint = endPoint.CreateTrackingPoint(PointTrackingMode.Negative);
+                var codeFixCategorySet = _suggestedActionCategoryRegistry.AllCodeFixes;
+
+                var session = _lightBulbBroker.CreateSession(codeFixCategorySet, textView, endTrackingPoint, endTrackingSpan, codeFixCategorySet, false);
```

Comment:
if you do this way, why do you use diagnostic analyzer and code fixer? if we have our own LB set up here, you can put suggested action directly rather than going thourhg diagnostic analyuzer/code fixer abstraction? using suggested action directly will let you don't need all the plumbing code required to use diagnostic analyzer or code fixer?

Reply:
It would certainly be my preference not to use the analyzer plumbing for this. From a little bit of digging it seems the light bulb broker can be influenced based on a supplied set of SuggestedActionCategories but then uses platform plumbing to get the list of fixes. If you know how I can better control the list of actions, I am open to changing this.

Reply:
yeah.  it seems unnecessary.  given that the information is stored in the buffer, it feels like we could just grab things out in a simpler way that doesn't involve needing a full analyzer/fixer.